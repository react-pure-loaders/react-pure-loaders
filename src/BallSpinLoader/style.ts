import { keyframes } from '@emotion/react';
import { globalAnimation } from '../mixins';
import { PRIMARY_COLOR } from '../variables';
import { Grid } from '../types';

const radius = '45px';

const ballSpin = keyframes({
  '100%': {
    opacity: 1,
  },
  '75%': {
    opacity: 0.2,
  },
});

function ballSpinLoader() {
  let i = 1;
  const grid: Grid = {};
  for (; i < 9; i += 1) {
    const quarter = (parseInt(radius, 10) / 2) + (parseInt(radius, 10) / 5.5);

    if (i === 1) {
      grid[`> div:nth-of-type(${i})`] = {
        animation: `${ballSpin} 2s ${(i * 0.9)} infinite linear`,
        left: 0,
        top: radius,
      };
    } else if (i === 2) {
      grid[`> div:nth-of-type(${i})`] = {
        animation: `${ballSpin} 2s ${(i * 0.9)}s infinite linear`,
        left: quarter,
        top: quarter,
      };
    } else if (i === 3) {
      grid[`> div:nth-of-type(${i})`] = {
        animation: `${ballSpin} 2s ${(i * 0.9)}s infinite linear`,
        left: radius,
        top: 0,
      };
    } else if (i === 4) {
      grid[`> div:nth-of-type(${i})`] = {
        animation: `${ballSpin} 2s ${(i * 0.9)}s infinite linear`,
        left: quarter,
        top: `-${quarter}px`,
      };
    } else if (i === 5) {
      grid[`> div:nth-of-type(${i})`] = {
        animation: `${ballSpin} 2s ${(i * 0.9)}s infinite linear`,
        left: 0,
        top: `-${radius}`,
      };
    } else if (i === 6) {
      grid[`> div:nth-of-type(${i})`] = {
        animation: `${ballSpin} 2s ${(i * 0.9)}s infinite linear`,
        left: `-${quarter}px`,
        top: `-${quarter}px`,
      };
    } else if (i === 7) {
      grid[`> div:nth-of-type(${i})`] = {
        animation: `${ballSpin} 2s ${(i * 0.9)}s infinite linear`,
        left: `-${radius}`,
        top: 0,
      };
    } else if (i === 8) {
      grid[`> div:nth-of-type(${i})`] = {
        animation: `${ballSpin} 2s ${(i * 0.9)}s infinite linear`,
        left: `-${quarter}px`,
        top: quarter,
      };
    }
  }

  return grid;
}

export default (color?: string) => ({
  ...ballSpinLoader(),
  position: 'relative' as 'relative',
  ' > div': {
    ...globalAnimation(),
    background: color || PRIMARY_COLOR,
    borderRadius: '100%',
    height: '15px',
    position: 'absolute' as 'absolute',
    width: '15px',
  },
});
