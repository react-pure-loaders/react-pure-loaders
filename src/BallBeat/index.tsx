/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallBeat({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
