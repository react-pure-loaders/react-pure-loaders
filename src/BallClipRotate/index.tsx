/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallClipRotate({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)} data-testid="ball-clip-rotate-container">
                <div/>
            </div>
    );
  }

  return null;
}
