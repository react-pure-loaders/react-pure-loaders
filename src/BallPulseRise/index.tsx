/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRise({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
