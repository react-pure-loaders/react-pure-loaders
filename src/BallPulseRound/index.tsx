/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRound({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
            </div>
    );
  }

  return null;
}
