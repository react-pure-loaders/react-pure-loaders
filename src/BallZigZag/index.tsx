/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZag({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
