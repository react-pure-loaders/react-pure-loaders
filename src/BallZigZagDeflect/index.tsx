/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZagDeflect({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
