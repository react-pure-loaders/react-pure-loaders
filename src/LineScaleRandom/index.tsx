/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
export default function LineScaleRandom({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div />
                <div />
                <div />
                <div />
            </div>
    );
  }

  return null;
}
