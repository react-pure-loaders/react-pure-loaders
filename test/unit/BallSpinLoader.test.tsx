import React from 'react';
import Chance from 'chance';
import { matchers } from '@emotion/jest';
import { render, cleanup } from '@testing-library/react';


import BallSpinLoader from '../../src/BallSpinLoader';
import { PRIMARY_COLOR } from '../../src/variables';

expect.extend(matchers);

const chance = new Chance();

describe('<BallSpinLoader>', () => {
    afterEach(cleanup);

    test('BallSpinLoader should match snapshot', () => {
        const { container } = render(<BallSpinLoader loading={true}/>);

        expect(container.firstChild).toMatchSnapshot();
    });

    test.skip('BallSpinLoader should have default color', () => {
        const { container } = render(<BallSpinLoader loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background', PRIMARY_COLOR, { target: '> div' });
    });

    test.skip('BallSpinLoader should have given color', () => {
        const color = chance.color({ format: 'hex' });
        const { container } = render(<BallSpinLoader color={color} loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background', color, { target: '> div' });
    });

    test('BallSpinLoader should have no children', () => {
        const { container } = render(<BallSpinLoader loading={false}/>);

        expect(container.firstChild).toBeNull();
    });

    test('BallSpinLoader should have three children', () => {
        const { container } = render(<BallSpinLoader loading={true}/>);

        expect(container.querySelectorAll('div')).toHaveLength(9);
    });
});
