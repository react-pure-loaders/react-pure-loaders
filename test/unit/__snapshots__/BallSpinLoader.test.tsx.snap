// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BallSpinLoader> BallSpinLoader should match snapshot 1`] = `
@keyframes animation-0 {
  100% {
    opacity: 1;
  }

  75% {
    opacity: 0.2;
  }
}

@keyframes animation-0 {
  100% {
    opacity: 1;
  }

  75% {
    opacity: 0.2;
  }
}

@keyframes animation-0 {
  100% {
    opacity: 1;
  }

  75% {
    opacity: 0.2;
  }
}

@keyframes animation-0 {
  100% {
    opacity: 1;
  }

  75% {
    opacity: 0.2;
  }
}

@keyframes animation-0 {
  100% {
    opacity: 1;
  }

  75% {
    opacity: 0.2;
  }
}

@keyframes animation-0 {
  100% {
    opacity: 1;
  }

  75% {
    opacity: 0.2;
  }
}

@keyframes animation-0 {
  100% {
    opacity: 1;
  }

  75% {
    opacity: 0.2;
  }
}

@keyframes animation-0 {
  100% {
    opacity: 1;
  }

  75% {
    opacity: 0.2;
  }
}

.emotion-0 {
  position: relative;
}

.emotion-0>div:nth-of-type(1) {
  -webkit-animation: animation-0 2s 0.9 infinite linear;
  animation: animation-0 2s 0.9 infinite linear;
  left: 0;
  top: 45px;
}

.emotion-0>div:nth-of-type(2) {
  -webkit-animation: animation-0 2s 1.8s infinite linear;
  animation: animation-0 2s 1.8s infinite linear;
  left: 30.68181818181818px;
  top: 30.68181818181818px;
}

.emotion-0>div:nth-of-type(3) {
  -webkit-animation: animation-0 2s 2.7s infinite linear;
  animation: animation-0 2s 2.7s infinite linear;
  left: 45px;
  top: 0;
}

.emotion-0>div:nth-of-type(4) {
  -webkit-animation: animation-0 2s 3.6s infinite linear;
  animation: animation-0 2s 3.6s infinite linear;
  left: 30.68181818181818px;
  top: -30.68181818181818px;
}

.emotion-0>div:nth-of-type(5) {
  -webkit-animation: animation-0 2s 4.5s infinite linear;
  animation: animation-0 2s 4.5s infinite linear;
  left: 0;
  top: -45px;
}

.emotion-0>div:nth-of-type(6) {
  -webkit-animation: animation-0 2s 5.4s infinite linear;
  animation: animation-0 2s 5.4s infinite linear;
  left: -30.68181818181818px;
  top: -30.68181818181818px;
}

.emotion-0>div:nth-of-type(7) {
  -webkit-animation: animation-0 2s 6.3s infinite linear;
  animation: animation-0 2s 6.3s infinite linear;
  left: -45px;
  top: 0;
}

.emotion-0>div:nth-of-type(8) {
  -webkit-animation: animation-0 2s 7.2s infinite linear;
  animation: animation-0 2s 7.2s infinite linear;
  left: -30.68181818181818px;
  top: 30.68181818181818px;
}

.emotion-0>div {
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  background: #ffffff;
  border-radius: 100%;
  height: 15px;
  position: absolute;
  width: 15px;
}

<div
  class="emotion-0"
>
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;
